gen/hooks.py

```diff
@@ -19,14 +19,31 @@
 # fmt: off
 
 def _gen_check(pname, ptype, strict=False):
+    SIGNED_CHECK = "isinstance({0}, int) and -1<<{1} <= {0} < 1<<{1}".format
```

Comment:
/me hates `.format` ... but whatever I suppose.

Reply:
Well, if it's the overhead you're worried about, none of this happens at runtime anyway! :stuck_out_tongue_winking_eye: